addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -313,73 +369,52 @@ const wSnippetMenu = weSnippetEditor.SnippetsMenu.extend({
-    },
+    }
     /**
      * @private
      * @return {Selection|null}
      */
     _getSelection() {
         return this.options.wysiwyg.odooEditor.document.getSelection();
-    },
+    }
     /**
      * @override
      */
     _addToolbar() {
-        this._super(...arguments);
-        this.$('#o_we_editor_toolbar_container > we-title > span').after($(`
-            <we-button class="fa fa-fw o_we_link o_we_highlight_animated_text d-none
-                ${this.$body.hasClass('o_animated_text_highlighted') ? 'fa-eye text-success' : 'fa-eye-slash'}"
-                title="${_t('Highlight Animated Text')}"
-                aria-label="Highlight Animated Text">
-            </we-button>
-        `));
-        this._toggleTextOptionsButton(".o_we_animate_text");
-        this._toggleHighlightAnimatedTextButton();
-        this._toggleTextOptionsButton(".o_we_text_highlight");
+        super._addToolbar(...arguments);
+        this.state.animatedTextHilighted = this.$body[0].classList.contains("o_animated_text_highlighted");
+        this.state.isTextAnimated = this._getTextOptionState(ANIMATED_TEXT_SELECTOR);
+        this.state.isTextHighlighted = this._getTextOptionState(HIGHLIGHTED_TEXT_SELECTOR);
 
         // As the toolbar displays css variable that are customizable by users,
         // we have the recompute the font size selector values.
         this.options.wysiwyg.odooEditor.computeFontSizeSelectorValues();
-    },
+    }
     /**
-     * Activates & deactivates the button used to add text options, depending
-     * on the selected element.
+     * Returns true if the selected text matches the selector.
      *
      * @private
      */
-    _toggleTextOptionsButton(selector) {
+    _getTextOptionState(textSelector) {
         if (!this._isValidSelection(this._getSelection())) {
             return;
         }
-        const textOptionsButton = this.el.querySelector(selector);
-        if (textOptionsButton) {
-            textOptionsButton.classList.toggle("active", !!this._getSelectedTextElement(textOptionsButton.dataset.textSelector));
-        }
-    },
-    /**
-     * Displays the button that allows to highlight the animated text if there
-     * is animated text in the page.
-     *
-     * @private
-     */
-    _toggleHighlightAnimatedTextButton() {
-        const $animatedText = this.getEditableArea().find('.o_animated_text');
-        this.$('#o_we_editor_toolbar_container .o_we_highlight_animated_text').toggleClass('d-none', !$animatedText.length);
-    },
+        return !!this._getSelectedTextElement(textSelector);
+    }
     /**
      * @private
      * @param {Node} node
      * @return {Boolean}
      */
     _isValidSelection(sel) {
         return sel.rangeCount && [...this.getEditableArea()].some(el => el.contains(sel.anchorNode));
-    },
+    }
     /**
      * @override
      */
     _isMobile() {
-        return wUtils.isMobile(this);
-    },
+        return this.websiteService.context.isMobile;
```

Comment:
I don't think this is an equivalent change, right? I have been meaning to unify those. Maybe this deserves its own commit?

Reply:
wUtils.isMobile calls `service_context_get` with a trigger_up (which needed to be removed) [1](https://github.com/odoo/odoo/blob/b7bf4cc2272f8e004e6fa2fa1f7f11c9d8e9ba8e/addons/website/static/src/js/utils.js#L408)

that trigger_up is handled by `WysiwygAdapter` and returns `{ "isMobile": this.websiteService.context.isMobile }` [2](https://github.com/odoo/odoo/blob/b7bf4cc2272f8e004e6fa2fa1f7f11c9d8e9ba8e/addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js#L1165)

Which I actually forgot to remove as I don't think the wysiwyg_adapter needs to handle it anymore.